--- build/patch-work/sun/nio/ch/EPollPoller.java.orig
+++ build/patch-work/sun/nio/ch/EPollPoller.java
@@ -25,6 +25,12 @@
 package sun.nio.ch;
 
 import java.io.IOException;
+import java.lang.foreign.Arena;
+import java.lang.foreign.FunctionDescriptor;
+import java.lang.foreign.Linker;
+import java.lang.foreign.SymbolLookup;
+import java.lang.invoke.MethodHandle;
+import java.nio.file.Path;
 import static sun.nio.ch.EPoll.*;
 
 /**
@@ -33,6 +39,13 @@
 
 class EPollPoller extends Poller {
     private static final int ENOENT = 2;
+    private static final MethodHandle NOOP_FN;
+    static {
+        Linker linker = Linker.nativeLinker();
+        NOOP_FN = linker.downcallHandle(
+            SymbolLookup.libraryLookup(Path.of("build/libnoop.so"), Arena.global()).find("noop").orElseThrow(),
+            FunctionDescriptor.ofVoid());
+    }
 
     private final int epfd;
     private final int event;
@@ -59,6 +72,7 @@
             err = EPoll.ctl(epfd, EPOLL_CTL_ADD, fdVal, (event | EPOLLONESHOT));
         if (err != 0)
             throw new IOException("epoll_ctl failed: " + err);
+        try { NOOP_FN.invoke(); } catch (Throwable t) { throw new RuntimeException(t); }
     }
 
     @Override
